### PR TITLE
Update EFS Driver Version to v1.5.4

### PR DESCRIPTION
**Pull request by jsitu777**
- update EFS Driver from v1.3.4 to v1.5.4 https://github.com/kubernetes-sigs/aws-efs-csi-driver/releases (https://github.com/kubernetes-sigs/aws-efs-csi-driver/tree/v1.5.4#deploy-the-driver)

### CODE REVIEW

File: tests/e2e/utils/auto-efs-setup.py [hunk 1]
```diff
@@ -184,7 +184,7 @@ def install_efs_driver():
     print("Installing EFS driver...")
 
     kubectl_kustomize_apply(
-        "https://github.com/kubernetes-sigs/aws-efs-csi-driver/deploy/kubernetes/overlays/stable/?ref=tags/v1.4.0"
+        "https://github.com/kubernetes-sigs/aws-efs-csi-driver/deploy/kubernetes/overlays/stable/?ref=release-1.5"
```
Comment (surajkota) [inline tests/e2e/utils/auto-efs-setup.py:187]: use tag?

**Review by surajkota — CHANGES_REQUESTED** [review]
need to configure terraform as well

File: tests/e2e/utils/auto-efs-setup.py (hunk 1 above)
Comment (jsitu777) [reply tests/e2e/utils/auto-efs-setup.py:187]: per the documentation this is the right way, and I tested it. https://github.com/kubernetes-sigs/aws-efs-csi-driver/tree/v1.5.4#deploy-the-driver

```
kubectl apply -k "github.com/kubernetes-sigs/aws-efs-csi-driver/deploy/kubernetes/overlays/stable/?ref=release-1.5"  

serviceaccount/efs-csi-controller-sa created
serviceaccount/efs-csi-node-sa created
clusterrole.rbac.authorization.k8s.io/efs-csi-external-provisioner-role created
clusterrolebinding.rbac.authorization.k8s.io/efs-csi-provisioner-binding created
deployment.apps/efs-csi-controller created
daemonset.apps/efs-csi-node created
csidriver.storage.k8s.io/efs.csi.aws.com configured
```

**Comment by jsitu777** [comment]
> need to configure terraform as well

https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/v4.28.0/modules/kubernetes-addons/aws-efs-csi-driver/main.tf

The 4.28 eks-blueprint updated `efs-csi-driver` helm version on Nov, 2022 https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/1191

File: tests/e2e/utils/auto-efs-setup.py (hunk 1 above)
Comment (jsitu777) [reply tests/e2e/utils/auto-efs-setup.py]: tag also works. changed.

**Review by surajkota — APPROVED** [review]

**Pull request merged**